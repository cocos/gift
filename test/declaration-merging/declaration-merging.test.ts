
import ps from 'path';
import { bundle } from '../../source/gift';

test('Declaration merging', () => {
    const inputPath = ps.join(__dirname, 'input.d.ts');
    const outputPath = ps.join(__dirname, 'output.d.ts');

    const result = bundle({
        input: inputPath,
        output: outputPath,
        entries: {
            'cc': 'index',
        },
    });

    expect(result).toMatchSnapshot();
});