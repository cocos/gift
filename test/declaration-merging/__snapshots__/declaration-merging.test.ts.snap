// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Declaration merging 1`] = `
Object {
  "groups": Array [
    Object {
      "code": "declare module \\"cc\\" {
    export namespace meow {
        export class Meow {
            /**
             * Main method.
             */
            public mainMethod(): void;
        }
    }
    export {};
}
",
      "parts": Object {
        "part1": Object {
          "code": "declare module \\"cc\\" {
    export namespace meow {
        interface Meow {
            /**
             * @part part1
             */
            partMethod(): void;
            /**
             * @part part1
             */
            partProperty: number;
        }
        namespace Meow {
            /**
             * @part part1
             */
            function partMethodStatic(): void;
            /**
             * @part part1
             */
            var partPropertyStatic: number;
        }
    }
}
",
          "path": "X:\\\\Dev\\\\Repos\\\\Leslie\\\\gift\\\\test\\\\declaration-merging\\\\output.part1.d.ts",
        },
      },
      "path": "X:\\\\Dev\\\\Repos\\\\Leslie\\\\gift\\\\test\\\\declaration-merging\\\\output.d.ts",
    },
  ],
}
`;
